pkg/controller/backup_controller.go

```diff
@@ -277,23 +272,23 @@ func (c *backupController) getLocationAndValidate(itm *api.Backup, defaultBackup
 	}
 	itm.Labels[api.StorageLocationLabel] = itm.Spec.StorageLocation
 
-	var backupLocation *api.BackupStorageLocation
-	backupLocation, err := c.backupLocationLister.BackupStorageLocations(itm.Namespace).Get(itm.Spec.StorageLocation)
-	if err != nil {
+	if storageLocation, err := c.backupLocationLister.BackupStorageLocations(itm.Namespace).Get(itm.Spec.StorageLocation); err != nil {
 		validationErrors = append(validationErrors, fmt.Sprintf("Error getting backup storage location: %v", err))
+	} else {
+		itm.StorageLocation = storageLocation
 	}
 
-	return backupLocation, validationErrors
+	return validationErrors
 }
 
 // defaultAndValidateSnapshotLocations ensures:
 // - each location name in Spec VolumeSnapshotLocation exists as a location
 // - exactly 1 location per existing or default provider
 // - a given default provider's location name is added to the Spec VolumeSnapshotLocation if it does not exist as a VSL
-func (c *backupController) defaultAndValidateSnapshotLocations(itm *api.Backup, defaultLocations map[string]string) []string {
-	var errors []string
-	perProviderLocationName := make(map[string]string)
-	var finalLocationNameList []string
+func (c *backupController) defaultAndValidateSnapshotLocations(itm *pkgbackup.Request, defaultLocations map[string]*api.VolumeSnapshotLocation) []string {
```

Comment:
Maybe rename so that it is clear we are calling this for the side effects that get applied to itm?